Rust/mechanics_fast/src/lib.rs

```diff
@@ -1,12 +1,28 @@
-use cgmath::{Vector3, InnerSpace};
+use cgmath::{InnerSpace, Vector3};
 
-#[no_mangle]
-pub extern "C" fn compute_gravitational_acceleration(displacement: Vector3<f64>, m2: f64) -> Vector3<f64> {
-    const GRAVITATIONAL_CONSTANT: f64 = 6.67430e-11;
+fn compute_gravitational_acceleration(displacement: Vector3<f32>, mass: f32) -> Vector3<f32> {
+    let g: f32 = 9.81;
 
-    let distance = displacement.magnitude();
+    let distance_squared = displacement.magnitude2();
 
-    let acceleration = displacement * (GRAVITATIONAL_CONSTANT * m2) / (distance * distance * distance);
+    if distance_squared > 0.0 {
+        let magnitude = g * mass / distance_squared.sqrt();
+        displacement.normalize() * magnitude
+    } else {
+        Vector3::new(0.0, 0.0, 0.0)
+    }
+}
+
+fn compute_acceleration(masses: &[f32], positions: &[Vector3<f32>], index_of_self: usize) -> Vector3<f32> {
```

Comment:
Could we make the floats 64-bit instead of 32-bit? That will make it compatible with the existing C# code.

```suggestion
fn compute_acceleration(masses: &[f64], positions: &[Vector3<f64>], index_of_self: usize) -> Vector3<f64> {
```